resources/sdk/pureclouddotnet/templates/ApiClient.mustache

```diff
@@ -156,6 +169,44 @@ namespace {{packageName}}.Client
             return request;
         }
 
+        private bool HasAccessTokenExpired(int statusCode)
```

Comment:
This method shouldn't be necessary. It doesn't matter if there _should_ be time left on the token; it only matters if the token was invalid. In the couple places this method is invoked, just check to see if the status code is 401 and invoke `HandleExpiredAccessToken()` if it is.